### PR TITLE
LPD-20495

### PR DESCRIPTION
**Pull request by jonathanmccann**
Story - https://liferay.atlassian.net/browse/LPD-6719
Subtask - https://liferay.atlassian.net/browse/LPD-20495

**Steps to Test**

1. Set `feature.flag.LPD-10945=true`
2. Go to Product Menu - Design - Page Templates - Display Page Templates
3. Add a new Display Page Template based on Blank with a Content Type of Calendar Event
4. Add a Paragraph fragment to the page
5. Map the text of the Paragraph fragment to one of the new fields
    - Calendar Name
    - Invitations
    - Repetitions
9. Publish the page
10. Click the three dot menu for the new Display Page Template and Mark as Default
11. Edit the home page
12. Add a Calendar widget to the page and publish
13. Add a new Calendar Event
14. Execute the following query in the database
    - `SELECT calendarBookingId FROM CalendarBooking;`
16. Copy the `calendarBookingId` and add it at the end of the following URL
    - `localhost:8080/h/`
18. Go to the URL

**Notes**

1. The [logic for the repetitions](https://github.com/liferay-content-management/liferay-portal/commit/2d678705a8c6fcaa2c70c463654255a6b4a8f3eb#diff-04b19c23905665e6dd1bf11cc7c819359843b0ef81b5bc8acb04f945fe13e2eeR231) was taken from [recurrence_util.js](https://github.com/liferay/liferay-portal/blob/master/modules/apps/calendar/calendar-web/src/main/resources/META-INF/resources/js/legacy/recurrence_util.js#L160)
    - I opted for duplicating the logic because the Javascript is called on the client-side so that whole mechanism would need to be refactored to send requests to the back end. It also seemed better to have duplicated logic rather than send requests on every change the user makes.
2. I do not like the [maps used for preparing the language keys](https://github.com/liferay-content-management/liferay-portal/commit/2d678705a8c6fcaa2c70c463654255a6b4a8f3eb#diff-04b19c23905665e6dd1bf11cc7c819359843b0ef81b5bc8acb04f945fe13e2eeR418) but I'm unsure of a different way to achieve the same goal. If you have any input I'd be happy to change it. This is also how it is done [in the frontend](https://github.com/liferay/liferay-portal/blob/master/modules/apps/calendar/calendar-web/src/main/resources/META-INF/resources/calendar_booking_recurrence_language_keys.jspf).
3. The [logic for determining the invitations](https://github.com/liferay-content-management/liferay-portal/commit/7a0d4f115ede00cf52edc680b098bb15e01614a0#diff-02e2c5a6a90f4e487c4c31e941e4297aad984ffcdf0b60394d8ef3ba092b443aR195) was taken [from the frontend](https://github.com/liferay/liferay-portal/blob/master/modules/apps/calendar/calendar-web/src/main/resources/META-INF/resources/edit_calendar_booking.jsp#L98-L108) as well. It looks like a lot of overhead for something that appears to be simple but it seems this is the way it is done.

If there are any questions or concerns please let me know.

### CODE REVIEW

**Comment by jonathanmccann** [comment]
ci:test:relevant

File: modules/apps/calendar/calendar-web/src/main/java/com/liferay/calendar/web/internal/info/item/provider/CalendarBookingInfoItemFieldValuesProvider.java [hunk 1]
```diff
@@ -67,13 +70,18 @@ calendarBooking, _getThemeDisplay())
 					calendarBooking.getCalendarBookingId())
 			).build();
 		}
+		catch (PortalException portalException) {
+			throw new RuntimeException(
+				"Caught unexpected exception", portalException);
+		}
```
Comment (adolfopa) [inline modules/apps/calendar/calendar-web/src/main/java/com/liferay/calendar/web/internal/info/item/provider/CalendarBookingInfoItemFieldValuesProvider.java:82]: Why do you need a catch for PortalException? It handles the error in the same way as the generic Exception.

**Review by adolfopa — CHANGES_REQUESTED** [review]

**Comment by adolfopa** [comment]
@jonathanmccann Only a very minor comment. Everything else looks great, so if there's a good reason to have the extra catch block, feel free to forward this directly.

File: modules/apps/calendar/calendar-web/src/main/java/com/liferay/calendar/web/internal/info/item/provider/CalendarBookingInfoItemFieldValuesProvider.java (hunk 1 above)
Comment (jonathanmccann) [reply modules/apps/calendar/calendar-web/src/main/java/com/liferay/calendar/web/internal/info/item/provider/CalendarBookingInfoItemFieldValuesProvider.java:82]: I was copying the pattern in other *FieldValuesProvider:

https://github.com/liferay/liferay-portal/blob/master/modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/info/item/provider/FileEntryInfoItemFieldValuesProvider.java#L89-L95

https://github.com/liferay/liferay-portal/blob/master/modules/apps/journal/journal-web/src/main/java/com/liferay/journal/web/internal/info/item/provider/JournalArticleInfoItemFieldValuesProvider.java#L117-L123

If it's not useful though I can remove it.

Comment (adolfopa) [reply modules/apps/calendar/calendar-web/src/main/java/com/liferay/calendar/web/internal/info/item/provider/CalendarBookingInfoItemFieldValuesProvider.java:82]: Sounds good

**Review by adolfopa — APPROVED** [review]

**Comment by adolfopa** [comment]
ci:forward

**Comment by austinchiang** [comment]
ci:forward

**Comment by AliciaGarciaGarcia** [comment]
Hi @jonathanmccann 

I think that semantic versioning is failing
image

**Comment by adolfopa** [comment]
See https://github.com/liferay-content-management/liferay-portal/pull/5290

**Pull request closed without merging**